io/eolearn/io/sentinelhub_service.py

```diff
@@ -132,6 +138,10 @@ def _add_data(self, eopatch, data):
         valid_mask = data[..., -1]
         data = data[..., :-1]
 
+        if self.image_format is None:
+            #image format was set to MimeType.TIFF_d16
+            data = data/2**16
```

Comment:
Shouldn't this be `data / 10 ** 4` (and also casting to `numpy.float32`)?
I have seen you updated a lot of unit tests but do you anywhere check if `image_format=None` returns the same result as `image_format=MimeType.TIFF_d32f`? I think such test would be mandatory.

Reply:
No. Sentinel-Hub OGC will always (as the last step in processing) scale values according to requested image type (uint8 for png/jpg/tiff, uint16 for tiff/depth=16 and float32 for tiff/depth=32f). Scaling is done as follows:
* int8: [0,1] -> [0, 256]
* int16: [0,1] -> [0, 2**16]
* float32: identity (no scaling)

The additional tests in class `TestImageFormats` all check that results are the "same"; not for identity, but rather for equality to some precision. Each test case does two requests, one in 32float and another with default `image_format` (`None`), and the results are compared.

Reply:
You are right about that. However I still have 2 concerns:

- Maximal value in 16-bit tiff is actually `65535 = 2 ** 16 - 1`. So I assume the scaling is done with [0,1] -> [0, 2**16 - 1]. That would mean that you have to divide by `2**16 - 1` instead of `2**16`. The differences between values you get from these 2 divisions have order of magnitude 1e-6 (I checked), so you won't see them in unit tests. But it still might be better to verify this with the core team. By the way, you can use `MimeType.TIFF_d16.get_expected_max_value()` to get the value `2 ** 16 - 1`.

- Sometimes reflectance values are greater than `1.0`. In that case if you request  data in `MimeType.TIFF_d32f`, will the service return you these original values `>1.0`? Because when requesting data in `MimeType.TIFF_d16` the values will be transformed down to `2**16 - 1` and that could cause inconsistencies between using `image_format=None` and `image_format=MimeType.TIFF_d32f`.

Sorry for complicating, but this is the most sensitive part of the PR and I would like us to be as consistent as possible.

Reply:
You are not complicating, that is the whole point of having PR and reviewing them. Thanks for the notes, will check and get back to you.

Reply:
The idea was to make a request in uint16, but then convert it to 0-1 (or better: to numbers that would be received using float32 request). 
Later on the data in eopatches would be (as is now) stores in 32bit floats.

Reply:
Yes, I understand. I recommend to review also an option where data would be stored in its native precision (int16) and only convert (and optionally scale) it on-the-fly when needed.